Tools/ardupilotwaf/mavgen.py

```diff
@@ -0,0 +1,49 @@
+#!/usr/bin/env python
+# encoding: utf-8
+# (c) Siddharth Bharat Purohit, 3DRobotics Inc.
+
+"""
+The **mavgen.py** program is a code generator which creates mavlink header files.
+"""
+
+from waflib import Task
+from waflib.TaskGen import extension
+import os
+
+class mavgen(Task.Task):
+	"""generate mavlink header files"""
+	color   = 'BLUE'
+	run_str = '${MAVGEN} --lang=C --wire-protocol=1.0 --output=generated_src/include/mavlink/v1.0/ ${SRC[0].abspath()}'
+
+@extension('.xml')
+def process_mavgen(self, node):
+	"""
+	Create a mavgen task
+	"""
+	tsk = self.create_task('mavgen', node, [])
+
+def configure(cfg):
+	"""
+	setup environment for mavlink header generator
+	"""
+	env = cfg.env
+	cfg.env.env = dict(os.environ)
+
+	cfg.start_msg('Checking for mavlink submodule')
+	if not cfg.srcnode.find_dir('modules/mavlink/'):
+		cfg.fatal('not initialized, please run: git submodule init && git submodule update')
+		return
+	cfg.end_msg('success')
+	env.MAVLINK_DIR = cfg.srcnode.find_dir('modules/mavlink/').abspath()
+
+	env.MAVLINK_HEADERS = cfg.bldnode.make_node('/generated_src/include/mavlink/v1.0/').abspath()
+
+	env.MAVGEN = env.MAVLINK_DIR  + '/pymavlink/tools/mavgen.py'
```

Comment:
> @guludo what do you mean by call python explicitly here?

See this https://github.com/bugobliterator/ardupilot/commit/6db2c5673c9f62df0e9f9160c3dae074e676c3d0#commitcomment-15400521